projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitActivity.java

```diff
@@ -55,124 +66,115 @@
   private String claimCaseSubscription =
       "למרות שח\"מ לא נתן את הסכמתו המפורשת מראש  \nלקבלת דבר/י הפרסומת";
 
-  EditText receivedDate;
-  DatePickerDialog datePickerDialog;
-  Calendar calendar;
-  Button createPdfButton;
-
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_lawsuit);
 
-    createPdfButton = findViewById(R.id.button_createPDF);
-    receivedDate = findViewById(R.id.editText_spamDate);
+    // Hold data of current lawsuit, set lawsuitProto values from form fields in FormFragment
+    lawsuitProto = Lawsuit.newBuilder().buildPartial();
+    selectedSmsSpam =
+        SmsMessage.newBuilder()
+            .setSender(getIntent().getExtras().getString("from"))
+            .setBody(getIntent().getExtras().getString("body"))
+            .setReceivedAt(getIntent().getExtras().getString("receivedAt"))
+            .build();
 
-    setTimeZones();
-    createPdfButton.setOnClickListener(
-        v -> {
-          checkPermissions();
-          if (permissionGranted) {
-            sharePdf(createPdf());
-          }
-        });
+    // Used for dates in the lawsuit form
+    DATE_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+    DATE_TIME_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
 
-    receivedDate.setText(getIntent().getExtras().getString("receivedAt"));
-    receivedDate.setOnClickListener(v -> displayDatePicker());
-  }
+    courtList = initCourts();
+    courtBuilder = new AlertDialog.Builder(LawsuitActivity.this);
 
-  // Used for date fields in the lawsuit
-  private void setTimeZones() {
-    DATE_TIME_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
-    DATE_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
-  }
+    viewPager = findViewById(R.id.pager);
+    PagerAdapter pagerAdapter = new FragmentPageAdapter(getSupportFragmentManager());
+    viewPager.setAdapter(pagerAdapter);
 
-  // Save PDF to device. Return absFilename of the file
-  private String createPdf() {
+    checkPermissions();
+  }
 
-    String template = null;
-    try {
-      template = fillTemplate();
-    } catch (IOException e) {
-      e.printStackTrace();
-    }
+  // Save PDF to device's external storage
+  void createPdf() {
```

Comment:
This can be moved to the form fragment since you are only using it there

Reply:
It's part of what I wrote in the channel, tell me what you think.